src/main/java/com/mercadolibre/bootcampmelifrescos/repository/BatchRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.mercadolibre.bootcampmelifrescos.repository;
+
+import com.mercadolibre.bootcampmelifrescos.model.Batch;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface BatchRepository extends JpaRepository<Batch, Long> {
+
+    @Query(value = "SELECT * FROM batch WHERE batch.product_id = :product_id AND batch.current_quantity > 0 ORDER BY batch.current_quantity DESC LIMIT 1" ,nativeQuery = true)
+    Batch getBatchByProductId(@Param("product_id") Long productId);
```

Comment:
Acho que daria para fazer algo tipo assim e não precisaria do @Query
```suggestion
    Optional<Batch> findByProduct(Product product);
```